std/range/primitives.d

```diff
@@ -2104,13 +2106,14 @@ if (isNarrowString!(C[]))
                  //Invalid UTF-8
                  msbs = 1;
              }
-             str = str[msbs .. $];
+             str = str.ptr[min(msbs, str.length) .. str.length];
```

Comment:
It's not necessary, but the implementation seems to care about avoiding bounds checking. Compare [lines 2097 and 2098](https://github.com/dlang/phobos/pull/4387/files/e1af1b0b51ea9f29d4ff8076d73c03ba10bfc73c#diff-9f63c74383984a09f5bf578493892e27R2097) above.